lib/os_io.c

```diff
@@ -5246,8 +5246,7 @@ ___stream_index *len_done;)
         {
           /* Check if the other part initiated a shutdown and we didn't reply */
```

Comment:
The test in the `if` is wrong (and was wrong before).  The code should be:
```
int shut = SSL_get_shutdown (d->tls);
if ((shut & SSL_RECEIVED_SHUTDOWN) && !(shut & SSL_SENT_SHUTDOWN) )
```
In other words "if we received a shutdown but not yet sent a shutdown then ...".

Reply:
Ah! Good catch, thanks. I updated the PR to use your code.